#! /bin/sh -e 
## dbm_ldopts.dpatch by Stephane Glondu <steph@glondu.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: add missing dynamic linking options for dbm (see OCaml PR#4699)

if [ $# -ne 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi
case "$1" in
    -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;	
    *)
	echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
	exit 1;;
esac

exit 0
@DPATCH@
diff --git a/otherlibs/dbm/Makefile b/otherlibs/dbm/Makefile
index 68fd200..894c822 100644
--- a/otherlibs/dbm/Makefile
+++ b/otherlibs/dbm/Makefile
@@ -21,6 +21,7 @@ CAMLOBJS=dbm.cmo
 COBJS=cldbm.o
 EXTRACFLAGS=$(DBM_INCLUDES)
 LINKOPTS=$(DBM_LINK)
+LDOPTS=-ldopt "$(DBM_LINK)"
 
 include ../Makefile
 
